api/http_api.md

```diff
@@ -0,0 +1,758 @@
+          "token_count": 0,
+          "update_date": "Thu, 29 Aug 2024 03:13:07 GMT",
+          "update_time": 1724901187843,
+          "vector_similarity_weight": 0.3
+        }
+    ],
+}
+```
+
+  
+The error response includes a JSON object like the following:
+
+```shell
+{
+    "code": 3016,
+    "message": "Can't access database to get the dataset list."
+}
+```
+
+## Upload files to a dataset
+
+**POST** `/api/v1/dataset/{dataset_id}/documents`
+
+Uploads files to a dataset. 
+
+### Request
+
+- Method: POST
+- URL: `/api/v1/dataset/{dataset_id}/documents`
+- Headers:
+  - `content-Type: application/json`
+  - 'Authorization: Bearer {YOUR_ACCESS_TOKEN}'
+- Body:
+  - `"dataset_name"`: `string`
+  - `"tenant_id"`: `string`
+  - `"embedding_model"`: `string`
+  - `"chunk_count"`: `integer`
+  - `"document_count"`: `integer`
+  - `"parse_method"`: `string`
+
+#### Request example
+
+```shell
+curl --request POST \
+     --url http://{address}//api/v1/dataset/{dataset_id}/documents \
+     --header 'Content-Type: multipart/form-data' \
+     --header 'Authorization: Bearer {YOUR_ACCESS_TOKEN}' \
+     --form 'dataset_id=ad403cd0758511efb63c0242ac120004' \      
+     --form 'file=@test.txt'
+```
+
+#### Request parameters
+
+- `"dataset_id"`: (*Body parameter*)
+    The dataset id
+- `"file"`: (*Body parameter*)  
+    The file to upload
+
+### Response
+
+The successful response includes a JSON object like the following:
+
+```shell
+{
+    "code": 0 
+}
+```
+
+- `"error_code"`: `integer`  
+  `0`: The operation succeeds.
+
+  
+The error response includes a JSON object like the following:
+
+```shell
+{
+    "code": 3016,
+    "message": "Can't connect database"
+}
+```
+
+## Download a file from a dataset
+
+**GET** `/api/v1/dataset/{dataset_id}/documents/{document_id}`
+
+Uploads files to a dataset. 
+
+### Request
+
+- Method: GET
+- URL: `/api/v1/dataset/{dataset_id}/documents/{document_id}`
+- Headers:
+  - `content-Type: application/json`
+  - 'Authorization: Bearer {YOUR_ACCESS_TOKEN}'
+
+#### Request example
+
+```shell
+curl --request GET \
+     --url http://{address}//api/v1/dataset/{dataset_id}/documents/{documents_id} \
+     --header 'Content-Type: application/json' \
+     --header 'Authorization: Bearer {YOUR_ACCESS_TOKEN}'
+```
+
+#### Request parameters
+
+- `"dataset_id"`: (*PATH parameter*)
+    The dataset id
+- `"documents_id"`: (*PATH parameter*)  
+    The document id of the file.
+
+### Response
+
+The successful response includes a JSON object like the following:
+
+```shell
+{
+    "code": 0 
+}
+```
+
+- `"error_code"`: `integer`  
+  `0`: The operation succeeds.
+
+  
+The error response includes a JSON object like the following:
+
+```shell
+{
+    "code": 3016,
+    "message": "Can't connect database"
+}
+```
+
+
+## List files of a dataset
+
+**GET** `/api/v1/dataset/{dataset_id}/documents?keywords={keyword}&page={page}&page_size={limit}&orderby={orderby}&desc={desc}`
```

Comment:
No document for `page` and `page_size`

Reply:
page and page_size is offset and limit to show in one page.